File: src/services/Services.NestJS/ollama-options.ts
export const OLLAMA_OPTIONS = 'OLLAMA_OPTIONS';

export interface OllamaOptions {
  baseUrl: string;
  defaultModel: string;
  timeoutMs: number;
}

export function createOllamaOptions(): OllamaOptions {
  return {
    baseUrl: process.env.OLLAMA_BASE_URL || 'http://localhost:11434',
    defaultModel: process.env.OLLAMA_DEFAULT_MODEL || 'llama3.2',
    timeoutMs: Number(process.env.OLLAMA_TIMEOUT_MS || 30000),
  };
}

File: src/services/Services.DotNet/OllamaOptions.cs
namespace Services.DotNet;

public class OllamaOptions
{
    public string BaseUrl { get; set; } = "http://localhost:11434";

    public string DefaultModel { get; set; } = "llama3.2";

    public TimeSpan Timeout { get; set; } = TimeSpan.FromSeconds(30);
}